fuel-p2p/src/behavior.rs

```diff
@@ -175,34 +194,42 @@ impl<Codec: NetworkCodec> FuelBehaviour<Codec> {
         &mut self,
         message_request: RequestMessage,
         peer_id: PeerId,
-        tx_channel: oneshot::Sender<Result<ResponseMessage, ReqResNetworkError>>,
+        channel_item: ResponseChannelItem,
     ) -> RequestId {
         let request_id = self
             .request_response
             .send_request(&peer_id, message_request);
 
-        self.outbound_requests_table.insert(request_id, tx_channel);
+        self.outbound_requests_table
+            .insert(request_id, channel_item);
 
         request_id
     }
 
     pub fn send_response_msg(
         &mut self,
         request_id: RequestId,
-        message: ResponseMessage,
+        message: OutboundResponse,
     ) -> Result<(), ResponseError> {
-        if let Some(channel) = self.inbound_requests_table.remove(&request_id) {
-            if self
-                .request_response
-                .send_response(channel, message)
-                .is_err()
-            {
-                debug!("Failed to send ResponseMessage for {:?}", request_id);
-                return Err(ResponseError::SendingResponseFailed);
+        match (
+            self.codec.convert_to_intermediate(&message),
+            self.inbound_requests_table.remove(&request_id),
+        ) {
+            (Ok(message), Some(channel)) => {
+                if self
+                    .request_response
+                    .send_response(channel, message)
+                    .is_err()
+                {
+                    debug!("Failed to send ResponseMessage for {:?}", request_id);
+                    return Err(ResponseError::SendingResponseFailed);
+                }
+            }
+            (Ok(_), None) => {
+                debug!("ResponseChannel for {:?} does not exist!", request_id);
+                return Err(ResponseError::ResponseChannelDoesNotExist);
             }
-        } else {
-            debug!("ResponseChannel for {:?} does not exist!", request_id);
-            return Err(ResponseError::ResponseChannelDoesNotExist);
+            _ => {}
```

Comment:
Should we log anything in the event of an error?

```suggestion
            (Err(e), channel) => { debug!("unexpected error occurred while processing message response: {:?}", e); }
```